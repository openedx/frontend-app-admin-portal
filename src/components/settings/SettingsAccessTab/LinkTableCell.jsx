import PropTypes from 'prop-types';
import { getConfig } from '@edx/frontend-platform/config';

const LinkTableCell = ({ row }) => {
  const { uuid } = row.original;
  return `${getConfig().ENTERPRISE_LEARNER_PORTAL_URL}/invite/${uuid}`;
};

LinkTableCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      uuid: PropTypes.string,
    }),
  }).isRequired,
};

export default LinkTableCell;
