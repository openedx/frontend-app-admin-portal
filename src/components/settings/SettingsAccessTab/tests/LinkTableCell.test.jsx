import React from 'react';
import renderer from 'react-test-renderer';

import LinkTableCell from '../LinkTableCell';

jest.mock('@edx/frontend-platform/config', () => ({
  getConfig: () => ({ ENTERPRISE_LEARNER_PORTAL_URL: 'http://localhost:8734' }),
}));

describe('LinkTableCell', () => {
  it('renders correctly', () => {
    const props = {
      row: {
        original: {
          uuid: 'test-invite-key-uuid',
        },
      },
    };
    const tree = renderer
      .create(<LinkTableCell {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
