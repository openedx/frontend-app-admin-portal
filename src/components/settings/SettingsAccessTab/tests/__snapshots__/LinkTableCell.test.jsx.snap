// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkTableCell renders correctly 1`] = `"http://localhost:8734/invite/test-invite-key-uuid"`;
